airflow/jobs/scheduler_job.py

```diff
@@ -1759,3 +1759,46 @@ def _emit_pool_metrics(self, session: Session = None) -> None:
     @provide_session
     def heartbeat_callback(self, session: Session = None) -> None:
         Stats.incr('scheduler_heartbeat', 1, 1)
+
+    @provide_session
+    def reset_state_for_orphaned_tasks(self, session: Session = None):
```

Comment:
I understand this is "overwriting" the same method in the parent class "BaseJob", but may be good to mention return type in the docstring (or add type annotations), especially given its return type (returns int) is different from the same method in the parent class (returns list).

Reply:
Ha, I just realise you have remove this method from `BaseJob`.

But still good to document down the return type I think.

Reply:
I should make both of them return the same type just for consistency though, good call.